pkg/services/app/add.go

```diff
@@ -317,36 +323,36 @@ func (a *App) commitAndPush(params AddParams, filters ...func(string) bool) erro
 	return nil
 }
 
-func (a *App) createAndUploadDeployKey(reposUrls []string, clusterName string, namespace string, dryRun bool) (string, error) {
-	secretRef := fmt.Sprintf("weave-gitops-%s", clusterName)
-	if dryRun {
-		return secretRef, nil
+func (a *App) createAndUploadDeployKey(repoUrl string, clusterName string, namespace string, dryRun bool) (string, error) {
+
+	if repoUrl == "" {
```

Comment:
Is this a valid case? Should we return an error if `repoUrl` is empty?

Reply:
I'm with you, I think there is a proper way to do this. I was trying to fix a failure in the unit tests. But that would involve a bit more changes. I prefer to have this merged first as I need to talk with @luizbafilho to clarify some doubts.

Reply:
not a valid case, that's left over the UX change, feel free to remove it.